### ISSUE TITLE
The `uitls.Masked` `__repr__` breaks with numpy 2.1.0

### ISSUE THREAD

**Issue by mwcraig** [astropy/astropy#16851]
### Description

Try to display the `__repr__` office manager anything that is a `utils.masked.Masked` under the hood fails with an import error in numpy 2.1.0.

### Expected behavior

The `__repr__` displays

### How to Reproduce

1. Install numpy 2.1.0
2. Run the code below
3. You get the traceback shown after the code snippet.

```python
from astropy.utils.masked import Masked
ma = Masked([1., 2., 3.], mask=[False, False, True])
ma
```

```python-traceback
ImportError                               Traceback (most recent call last)
File ~/mambaforge/envs/ccdp-dev-py312/lib/python3.12/site-packages/IPython/core/formatters.py:711, in PlainTextFormatter.__call__(self, obj)
    704 stream = StringIO()
    705 printer = pretty.RepresentationPrinter(stream, self.verbose,
    706     self.max_width, self.newline,
    707     max_seq_length=self.max_seq_length,
    708     singleton_pprinters=self.singleton_printers,
    709     type_pprinters=self.type_printers,
    710     deferred_pprinters=self.deferred_printers)
--> 711 printer.pretty(obj)
    712 printer.flush()
    713 return stream.getvalue()

File ~/mambaforge/envs/ccdp-dev-py312/lib/python3.12/site-packages/IPython/lib/pretty.py:419, in RepresentationPrinter.pretty(self, obj)
    408                         return meth(obj, self, cycle)
    409                 if (
    410                     cls is not object
    411                     # check if cls defines __repr__
   (...)
    417                     and callable(_safe_getattr(cls, "__repr__", None))
    418                 ):
--> 419                     return _repr_pprint(obj, self, cycle)
    421     return _default_pprint(obj, self, cycle)
    422 finally:

File ~/mambaforge/envs/ccdp-dev-py312/lib/python3.12/site-packages/IPython/lib/pretty.py:787, in _repr_pprint(obj, p, cycle)
    785 """A pprint that just redirects to the normal repr function."""
    786 # Find newlines and replace them with p.break_()
--> 787 output = repr(obj)
    788 lines = output.splitlines()
    789 with p.group():

File ~/mambaforge/envs/ccdp-dev-py312/lib/python3.12/site-packages/astropy/utils/masked/core.py:1294, in MaskedNDArray.__repr__(self)
   1293 def __repr__(self):
-> 1294     return np.array_repr(self)

File ~/mambaforge/envs/ccdp-dev-py312/lib/python3.12/site-packages/astropy/utils/masked/core.py:937, in MaskedNDArray.__array_function__(self, function, types, args, kwargs)
    933 def __array_function__(self, function, types, args, kwargs):
    934     # TODO: go through functions systematically to see which ones
    935     # work and/or can be supported.
    936     if function in MASKED_SAFE_FUNCTIONS:
--> 937         return super().__array_function__(function, types, args, kwargs)
    939     elif function in APPLY_TO_BOTH_FUNCTIONS:
    940         helper = APPLY_TO_BOTH_FUNCTIONS[function]

File ~/mambaforge/envs/ccdp-dev-py312/lib/python3.12/site-packages/numpy/_core/arrayprint.py:1655, in array_repr(arr, max_line_width, precision, suppress_small)
   1609 @array_function_dispatch(_array_repr_dispatcher, module='numpy')
   1610 def array_repr(arr, max_line_width=None, precision=None, suppress_small=None):
   1611     """
   1612     Return the string representation of an array.
   1613
   (...)
   1653
   1654     """
-> 1655     return _array_repr_implementation(
   1656         arr, max_line_width, precision, suppress_small)

File ~/mambaforge/envs/ccdp-dev-py312/lib/python3.12/site-packages/numpy/_core/arrayprint.py:1578, in _array_repr_implementation(arr, max_line_width, precision, suppress_small, array2string)
   1576     lst = repr(arr.item())
   1577 elif arr.size > 0 or arr.shape == (0,):
-> 1578     lst = array2string(arr, max_line_width, precision, suppress_small,
   1579                        ', ', prefix, suffix=suffix)
   1580 else:  # show zero-length shape unless it is (0,)
   1581     lst = "[], shape=%s" % (repr(arr.shape),)

File ~/mambaforge/envs/ccdp-dev-py312/lib/python3.12/site-packages/astropy/utils/masked/core.py:960, in MaskedNDArray.__array_function__(self, function, types, args, kwargs)
    958 dispatched_function = DISPATCHED_FUNCTIONS[function]
    959 try:
--> 960     dispatched_result = dispatched_function(*args, **kwargs)
    961 except NotImplementedError:
    962     return self._not_implemented_or_raise(function, types)

File ~/mambaforge/envs/ccdp-dev-py312/lib/python3.12/site-packages/astropy/utils/masked/function_helpers.py:1232, in array2string(a, max_line_width, precision, suppress_small, separator, prefix, style, formatter, threshold, edgeitems, sign, floatmode, suffix, legacy)
   1230     from numpy.core.arrayprint import _format_options, _make_options_dict
   1231 else:
-> 1232     from numpy._core.arrayprint import _format_options, _make_options_dict
   1234 overrides = _make_options_dict(
   1235     precision,
   1236     threshold,
   (...)
   1244     floatmode,
   1245 )
   1246 options = _format_options.copy()

ImportError: cannot import name '_format_options' from 'numpy._core.arrayprint' (/Users/mattcraig/mambaforge/envs/ccdp-dev-py312/lib/python3.12/site-packages/numpy/_core/arrayprint.py)
```

### Versions

```python
import astropy
try:
    astropy.system_info()
except AttributeError:
    import platform; print(platform.platform())
    import sys; print("Python", sys.version)
    import astropy; print("astropy", astropy.__version__)
    import numpy; print("Numpy", numpy.__version__)
    import erfa; print("pyerfa", erfa.__version__)
    try:
        import scipy
        print("Scipy", scipy.__version__)
    except ImportError:
        print("Scipy not installed")
    try:
        import matplotlib
        print("Matplotlib", matplotlib.__version__)
    except ImportError:
        print("Matplotlib not installed")
```
```
# Paste the result here
macOS-14.6.1-arm64-arm-64bit
Python 3.12.5 | packaged by conda-forge | (main, Aug  8 2024, 18:32:50) [Clang 16.0.6 ]
astropy 6.1.2
Numpy 2.1.0
pyerfa 2.0.1.4
Scipy 1.14.0
Matplotlib 3.9.1.post1

```

**Comment by neutrinoceros** [comment]
This is already fixed on main (in #16755), but the milestone on that PR is misleading: it was actually backported but only after the 6.1.2 tag was cut.

**Comment by mwcraig** [comment]
Ah, thanks, didn't see that. Closing since it is fixed.
